common/cmake/devices.cmake

```diff
@@ -355,7 +364,8 @@ function(DEFINE_DEVICE_TYPE)
     ${DEFINE_DEVICE_TYPE_DEVICE_TYPE}
     PROPERTIES
     DEVICE_MERGED_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${FINAL_OUTPUT}
-  )
+    )
+  add_dependencies(${DEFINE_DEVICE_TYPE_DEVICE_TYPE} ${FINAL_TARGET})
```

Comment:
Use "append_file_dependencies(${FINAL_OUTPUT})" over add_dependencies.

Reply:
That appends to a list, but I'd still need to add the dependency right?

Reply:
So if you only depend on FINAL_TARGET, you have the target dependency but not the file dependency.  As a result, if the FINAL_TARGET was already built, but FINAL_OUTPUT changed, the device type won't be see as "old".

CMake's depedency system for output is crazy, which is why I recommend using "append_file_dependencies".

Reply:
For further reading: https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/

Reply:
Done